api.py

```
@@ -0,0 +1,501 @@
+    safe_ticker = normalize_ticker(ticker)
+
+    if not safe_ticker:
+        raise ValueError("Ticker is required.")
+
+    safe_ticker = "".join(
+        char for char in safe_ticker
+        if char.isalnum() or char in ("_", "-")
+    )
+
+    if not safe_ticker:
+        raise ValueError("Invalid ticker.")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)